ibm/resource_ibm_cloudant.go

```diff
@@ -0,0 +1,586 @@
+		if err != nil {
+			return fmt.Errorf("Error updating activity tracker events: %s", err)
+		}
+	}
+
+	if d.HasChange("capacity") {
+		err := updateCloudantInstanceCapacity(client, d)
+		if err != nil {
+			return fmt.Errorf("Error retrieving capacity throughput information: %s", err)
+		}
+	}
+
+	if d.HasChange("enable_cors") {
+		err := updateCloudantInstanceCors(client, d)
+		if err != nil {
+			return fmt.Errorf("Error updating CORS settings: %s", err)
+		}
+	}
+
+	return resourceIBMCloudantRead(d, meta)
+}
+
+func setCloudantLegacyCredentials(d *schema.ResourceData, meta interface{}) error {
+	rsConClient, err := meta.(ClientSession).ResourceControllerV2API()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Id()
+	resourceInstanceGet := rc.GetResourceInstanceOptions{
+		ID: &instanceID,
+	}
+
+	instance, response, err := rsConClient.GetResourceInstance(&resourceInstanceGet)
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving resource instance: %s\n%s", err, response)
+		return err
+	}
+
+	parameters := instance.Parameters
+
+	if legacyCredentials, ok := parameters["legacyCredentials"]; ok {
+		switch v := legacyCredentials.(type) {
+		case bool:
+			d.Set("legacy_credentials", v)
+		case string:
+			d.Set("legacy_credentials", v == "true")
+		default:
+			d.Set("legacy_credentials", false)
+		}
+	}
+
+	if environmentCRN, ok := parameters["environment_crn"]; ok {
+		v := environmentCRN.(string)
+		d.Set("environment_crn", v)
+	}
+
+	return nil
+}
+
+func setCloudantResourceControllerURL(d *schema.ResourceData, meta interface{}) error {
+	crn := d.Get(ResourceCRN).(string)
+	rcontroller, err := getBaseController(meta)
+	if err != nil {
+		return err
+	}
+	d.Set(ResourceControllerURL, rcontroller+"/services/cloudantnosqldb/"+url.QueryEscape(crn))
+
+	return nil
+}
+
+func getCloudantClient(d *schema.ResourceData, meta interface{}) (*cloudantv1.CloudantV1, error) {
+
+	session, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoint string
+	extensions := d.Get("extensions").(map[string]interface{})
+	if v, ok := extensions["endpoints.public"]; ok {
+		endpoint = "https://" + v.(string)
+	}
+
+	switch session.Config.Visibility {
+	case "private":
+		_, ok := extensions["endpoints.private"]
+		if !ok {
+			return nil, fmt.Errorf("Missing endpoints.private in extensions")
+		}
+		endpoint = "https://" + extensions["endpoints.private"].(string)
+	case "public-and-private":
+		if v, ok := extensions["endpoints.private"]; ok {
+			endpoint = "https://" + v.(string)
+		}
+	}
+
+	endpoint = envFallBack([]string{"IBMCLOUD_CLOUDANT_ENDPOINT"}, endpoint)
+	if endpoint == "" {
+		return nil, fmt.Errorf("Missing endpoints.public in extensions")
+	}
+
+	token := session.Config.IAMAccessToken
+	token = strings.Replace(token, "Bearer ", "", -1)
+
+	client, err := cloudantv1.NewCloudantV1(&cloudantv1.CloudantV1Options{
+		Authenticator: &core.BearerTokenAuthenticator{
+			BearerToken: token,
```

Comment:
Please include apikey authentication as well
something similar to https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/config.go#L1161